/*
 *  Copyright (c) 2025 The Android Open Source Project
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at:
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

syntax = "proto3";

package bluetooth_hal.config.proto;

message FirmwareConfig {
  optional string firmware_folder_name = 1;
  optional string firmware_file_name = 2;
  optional uint32 chip_id = 3;
  optional uint32 load_mini_drv_delay_ms = 4;
  optional uint32 launch_ram_delay_ms = 5;

  enum FirmwareDataLoadingType {
    PACKET_BY_PACKET = 0;
    ACCUMULATED_BUFFER = 1;
  }
  optional FirmwareDataLoadingType firmware_data_loading_type = 6;

  optional SetupCommands setup_commands = 7;
}

message SetupCommands {
  repeated uint32 hci_reset = 1;
  repeated uint32 hci_read_chip_id = 2;
  repeated uint32 hci_update_chip_baud_rate = 3;
  repeated uint32 hci_set_fast_download = 4;
  repeated uint32 hci_download_minidrv = 5;
  repeated uint32 hci_vsc_launch_ram = 6;
  repeated uint32 hci_read_fw_version = 7;
  repeated uint32 hci_setup_low_power_mode = 8;
  repeated uint32 hci_write_bd_address = 9;
}
